recipes/swig/all/conanfile.py

```diff
@@ -83,8 +91,8 @@ def _configure_autotools(self):
 
         libargs = list("-L\"{}\"".format(p) for p in deps_libpaths) + list("-l\"{}\"".format(l) for l in deps_libs)
         args = [
-            "PCRE_LIBS={}".format(" ".join(libargs)),
-            "PCRE_CPPFLAGS={}".format(" ".join("-D{}".format(define) for define in deps_defines)),
+            "{}_LIBS={}".format("PCRE2" if self._use_pcre2 else "PCRE", " ".join(libargs)),
+            "{}_CPPFLAGS={}".format("PCRE2" if self._use_pcre2 else "PCRE", " ".join("-D{}".format(define) for define in deps_defines)),
```

Comment:
Pass the right flags.

Note that I think swig has always used `PCRE_CFLAGS` (now `PCRE2_CFLAGS`), not `PCRE_CPPFLAGS`

cf https://github.com/swig/swig/blob/efe5f181cf69975cf2fa1dcddefc29b93e4a6196/configure.ac#L71